instrumentation/opentelemetry-instrumentation-tortoiseorm/src/opentelemetry/instrumentation/tortoiseorm/__init__.py

```diff
@@ -0,0 +1,316 @@
+        )
+
+    def _hydrate_span_from_args(self, connection, query, parameters) -> dict:
+        """Get network and database attributes from connection."""
+        span_attributes = {}
+        capabilities = getattr(connection, "capabilities", None)
+        if capabilities:
+            if capabilities.dialect == "sqlite":
+                span_attributes[
+                    SpanAttributes.DB_SYSTEM
+                ] = DbSystemValues.SQLITE.value
+            elif capabilities.dialect == "postgres":
+                span_attributes[
+                    SpanAttributes.DB_SYSTEM
+                ] = DbSystemValues.POSTGRESQL.value
+            elif capabilities.dialect == "mysql":
+                span_attributes[
+                    SpanAttributes.DB_SYSTEM
+                ] = DbSystemValues.MYSQL.value
+        dbname = getattr(connection, "filename", None)
+        if dbname:
+            span_attributes[SpanAttributes.DB_NAME] = dbname
+        dbname = getattr(connection, "database", None)
+        if dbname:
+            span_attributes[SpanAttributes.DB_NAME] = dbname
+        if query is not None:
+            span_attributes[SpanAttributes.DB_STATEMENT] = query
+        user = getattr(connection, "user", None)
+        if user:
+            span_attributes[SpanAttributes.DB_USER] = user
+        host = getattr(connection, "host", None)
+        if host:
+            span_attributes[SpanAttributes.NET_PEER_NAME] = host
+        port = getattr(connection, "port", None)
+        if port:
+            span_attributes[SpanAttributes.NET_PEER_PORT] = port
+
+        if self.capture_parameters:
+            if parameters is not None and len(parameters) > 0:
+                span_attributes["db.statement.parameters"] = str(parameters)
+
+        return span_attributes
+
+    async def _do_execute(self, func, instance, args, kwargs):
+
+        exception = None
+        name = args[0].split()[0]
+
+        with self._tracer.start_as_current_span(
+            name, kind=SpanKind.CLIENT
+        ) as span:
+            if span.is_recording():
+                span_attributes = self._hydrate_span_from_args(
+                    instance,
+                    args[0],
+                    args[1:],
+                )
+                for attribute, value in span_attributes.items():
+                    span.set_attribute(attribute, value)
+
+            try:
+                result = await func(*args, **kwargs)
+            except Exception as exc:  # pylint: disable=W0703
+                exception = exc
+                raise
+            finally:
+                if span.is_recording() and exception is not None:
+                    span.set_status(Status(StatusCode.ERROR))
+
+        return result
+
+    async def _from_queryset(self, func, modelcls, args, kwargs):
+
+        exception = None
+        name = "pydantic.{0}".format(func.__name__)
+
+        with self._tracer.start_as_current_span(
```

Comment:
When are these `INTERNAL` spans created? What are they representing?

Reply:
conversion of the database responses into pydantic models. This takes 90% of the execution time and is something you'd want to measure if you're diagnosing slow response times.